frontend/src/v5/ui/routes/viewer/toolbar/buttons/toolbarButtons.component.tsx

```diff
@@ -94,3 +97,34 @@ export const CalibrationButton = () => {
 	);
 };
 
+export const VerticalCalibrationButton = () => {
+	const { isCalibratingPlanes, setIsCalibratingPlanes, step } = useContext(CalibrationContext);
+	return (
+		<ToolbarButton
+			Icon={VerticalCalibrationIcon}
+			hidden={step !== 2}
+			selected={isCalibratingPlanes}
+			onClick={() => setIsCalibratingPlanes(!isCalibratingPlanes)}
+			title={formatMessage({ id: 'viewer.toolbar.icon.verticalCalibration', defaultMessage: 'Vertical Calibration' })}
+		/>
+	);
+};
+
+export const VerticalRange = () => {
+	const { verticalPlanes, isCalibratingPlanes } = useContext(CalibrationContext);
+	const unit = ModelHooksSelectors.selectUnit();
+	const conversionFactor = unit === 'ft' ? 1 : UNITS_CONVERSION_FACTORS_TO_METRES[unit];
+	const rangeValue = ((verticalPlanes?.[1] - verticalPlanes?.[0]) / conversionFactor).toFixed(2);
+	const unitLabel = unit === 'ft' ? formatMessage({
+		id: 'viewer.toolbar.icon.verticalRange.imperial',
+		defaultMessage: 'ft',
+	}) : formatMessage({
+		id: 'viewer.toolbar.icon.verticalRange.metric',
+		defaultMessage: 'm',
+	});
```

Comment:
I've done this, but I think we should address some bits related to units in a new issue.

For instance there are a few places where we use the CONTAINER_TYPE value in the UI which means it is an unformatted string. We should use the new abbreviation keys.

In settingsModal.component we have a similar array called UNITS which only contains formatted messages (there is also a redundant CONTAINER_TYPES array). I'm pretty sure this means it would break if a translation were used as the Select input is using a formatted message for its `value` which wouldn't match the backend values

Also I'm not sure this data structure is the best. For example It would be easier to be able to do something like `CONTAINER_UNITS.ft.abbreviation` than `CONTAINER_UNITS.find(({ value }) => value === 'ft').abbreviation`

Reply:
Also the fact that it's called CONTAINER_UNITS but we're also using it for federations and drawings